### ISSUE TITLE
AttributeError: module 'torch' has no attribute 'bucketize'

### ISSUE THREAD

**Issue by loretoparisi** [ming024/FastSpeech2#7]
I get the following error:

```
root@75adae8f35d1:/app# python3 synthesize.py --step 300000
|{DH AH0 N EY1 SH AH0 N Z T UH1 R IH2 Z AH0 M M IH1 N AH0 S T ER0 HH AE1 Z AO1 L S OW0 EH0 N K ER1 IH0 JH D AO2 S T R EY1 L Y AH0 N Z T UW1 T EY1 K DH EH1 R HH AA1 L AH0 D EY2 Z W IH0 DH IH1 N DH AH0 K AH1 N T R IY0 DH IH1 S Y IH1 R} |
Traceback (most recent call last):
  File "synthesize.py", line 94, in <module>
    synthesize(model, text, sentence, prefix='step_{}'.format(args.step))
  File "synthesize.py", line 48, in synthesize
    mel, mel_postnet, duration_output, f0_output, energy_output = model(text, src_pos)
  File "/usr/local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/torch/nn/parallel/data_parallel.py", line 143, in forward
    return self.module(*inputs, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/app/fastspeech2.py", line 33, in forward
    encoder_output, d_target, p_target, e_target, max_length)
  File "/usr/local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "/app/modules.py", line 47, in forward
    pitch_embedding = self.pitch_embedding(torch.bucketize(pitch_prediction, self.pitch_bins))
AttributeError: module 'torch' has no attribute 'bucketize'
```

I'm running in CPU, and I had to modify `get_FastSpeech2` like this:

```python
def get_FastSpeech2(num):
    checkpoint_path = os.path.join(hp.checkpoint_path, "checkpoint_{}.pth.tar".format(num))
    model = nn.DataParallel(FastSpeech2())
    if torch.cuda.is_available():
        model.load_state_dict(torch.load(checkpoint_path)['model'])
    else:
        model.load_state_dict(torch.load(checkpoint_path, map_location=torch.device('cpu'))['model'])
    model.requires_grad = False
    model.eval()
    return model
```

to set `map_location` to cpu device.

**Comment by ming024** [comment]
@loretoparisi What is your PyTorch version? This functionality is newly introduced in PyTorch 1.6, which is still not stable yet.

**Comment by loretoparisi** [comment]
@ming024 got it, I was using `1.5.0` but someone told me I have to install

```

**Comment by loretoparisi** [comment]
@ming024 I have fixed the error using `1.6.0dev`, but now I'm getting

```
Traceback (most recent call last):
  File "synthesize.py", line 94, in <module>
    synthesize(model, text, sentence, prefix='step_{}'.format(args.step))
  File "synthesize.py", line 61, in synthesize
    Audio.tools.inv_mel_spec(mel_postnet, os.path.join(hp.test_path, '{}_griffin_lim_{}.wav'.format(prefix, sentence)))
  File "/app/audio/tools.py", line 63, in inv_mel_spec
    spec_from_mel[:, :, :-1]), _stft.stft_fn, griffin_iters)
  File "/app/audio/audio_processing.py", line 74, in griffin_lim
    _, angles = stft_fn.transform(signal)
  File "/app/audio/stft.py", line 66, in transform
    input_data.cuda(),
  File "/usr/local/lib/python3.7/site-packages/torch/cuda/__init__.py", line 150, in _lazy_init
    _check_driver()
  File "/usr/local/lib/python3.7/site-packages/torch/cuda/__init__.py", line 54, in _check_driver
    http://www.nvidia.com/Download/index.aspx""")
AssertionError: 
Found no NVIDIA driver on your system. Please check that you
have an NVIDIA GPU and installed a driver from
http://www.nvidia.com/Download/index.aspx
```

I'm using CPU.

**Comment by loretoparisi** [comment]
**[UPDATE]**
Ok I have fixed the previous one in `sftp.py` doing in `def transform(self, input_data):` like:

```python
if torch.cuda.is_available():
            forward_transform = F.conv1d(
                input_data.cuda(),
                Variable(self.forward_basis, requires_grad=False).cuda(),
                stride=self.hop_length,
                padding=0).cpu()
        else:
            forward_transform = F.conv1d(
                input_data,
                Variable(self.forward_basis, requires_grad=False),
                stride=self.hop_length,
                padding=0).cpu()
```

the problem is that it seems that the waveglov
```

**Comment by ming024** [comment]
@loretoparisi it is because in the MelGAN implementation the weight normalization parameters are removed when the model is set to inference mode. 
https://github.com/seungwonpark/melgan/blob/aca59909f6dd028ec808f987b154535a7ca3400c/hubconf.py#L22

You can see this [colab](https://colab.research.google.com/drive/1S60pytpB1OcEFrd-SkYyjtBsBHYepRSG?usp=sharing) for a simple solution.

**Comment by ming024** [comment]
closed #7
